src/plugins/explore/public/components/visualizations/table/table_vis.tsx

```diff
@@ -25,8 +25,13 @@ interface TableVisProps {
 
 export const TableVis = React.memo(
   ({ rows, columns, styleOptions, pageSizeOptions, showStyleSelector }: TableVisProps) => {
+    const sortedColumns = useMemo(() => columns.sort((a, b) => a.id - b.id), [columns]);
```

Comment:
Sort will update the original columns, I prefer to construct a new array for sorting.